app/helpers/fhir_helper.rb

```diff
@@ -107,57 +112,67 @@ def races_as_hash(patient)
-      foreign_address_line_1: foreign_address&.line&.first,
-      foreign_address_line_2: foreign_address&.line&.second,
-      foreign_address_line_3: foreign_address&.line&.third,
-      foreign_address_city: foreign_address&.city,
-      foreign_address_state: foreign_address&.state,
-      foreign_address_zip: foreign_address&.postalCode,
-      foreign_address_country: foreign_address&.country,
-      primary_language: patient&.communication&.first&.language&.coding&.first&.display,
-      interpretation_required: patient&.communication&.first&.preferred,
-      white: race_code?(patient, '2106-3', 'ombCategory', false),
-      black_or_african_american: race_code?(patient, '2054-5', 'ombCategory', false),
-      american_indian_or_alaska_native: race_code?(patient, '1002-5', 'ombCategory', false),
-      asian: race_code?(patient, '2028-9', 'ombCategory', false),
-      native_hawaiian_or_other_pacific_islander: race_code?(patient, '2076-8', 'ombCategory', false),
-      race_other: race_code?(patient, '2131-1', 'detailed', false),
-      race_unknown: race_code?(patient, 'unknown', 'http://hl7.org/fhir/StructureDefinition/data-absent-reason', true),
-      race_refused_to_answer: race_code?(patient, 'asked-declined', 'http://hl7.org/fhir/StructureDefinition/data-absent-reason', true),
+      monitoring: { value: patient&.active.nil? ? false : patient.active, path: 'Patient.active' },
+      first_name: { value: patient&.name&.first&.given&.first, path: 'Patient.name[0].given[0]' },
+      middle_name: { value: patient&.name&.first&.given&.second, path: 'Patient.name[0].given[1]' },
+      last_name: { value: patient&.name&.first&.family, path: 'Patient.name[0].family' },
+      primary_telephone: { value: from_fhir_phone_number(primary_phone&.value), path: "Patient.telecom[#{patient&.telecom&.index(primary_phone)}].value" },
+      secondary_telephone: { value: from_fhir_phone_number(secondary_phone&.value),
+                             path: "Patient.telecom[#{patient&.telecom&.index(secondary_phone)}].value" },
+      email: { value: email&.value, path: "Patient.telecom[#{patient&.telecom&.index(email)}].value" },
+      date_of_birth: { value: patient&.birthDate, path: 'Patient.birthDate' },
+      age: { value: Patient.calc_current_age_fhir(patient&.birthDate), path: 'Patient.birthDate' },
+      address_line_1: { value: address&.line&.first, path: "Patient.address[#{address_index}].line[0]" },
+      address_line_2: { value: address&.line&.second, path: "Patient.address[#{address_index}].line[1]" },
+      address_city: { value: address&.city, path: "Patient.address[#{address_index}].city" },
+      address_county: { value: address&.district, path: "Patient.address[#{address_index}].district" },
+      address_state: { value: address&.state, path: "Patient.address[#{address_index}].state" },
+      address_zip: { value: address&.postalCode, path: "Patient.address[#{address_index}].postalCode" },
```

Comment:
In the case where I send the following as an `address`:
```json
    "address": [
        {
            "extension": [
                {
                    "url": "http://saraalert.org/StructureDefinition/address-type",
                    "valueString": "Foreign"
                }
            ],
            "line": [
                "7653 d ff",
                "Suite 532",
                "Suite 690"
            ]
        }
    ]
```
the following error is reported:
```
        {
            "severity": "error",
            "code": "processing",
            "diagnostics": "'State' is required unless a \"Foreign Address Country\" is specified",
            "expression": [
                "Patient.address[].state"
            ]
        }
```
Because we either require `address_state` or `foreign_address_country` to be specified. Since no domestic address is given, we end up with the empty brackets. I'm not sure entirely the best approach for handling this. We could make it so that `address_index` is set to `foreign_address_index` if `address_index` is `nil`.

Reply:
Good call, and this made me realize a bug we had. Since the `address_state` validation rule depends on `foreign_address_country`, but were mapping address _before_ foreign address, it would erroneously fail validation if the request has an address without a state and a foreign address with a country. I'm correcting this by mapping foreign address first, and also applying your suggestion for the FHIR Path, with an additional fallback to `0` for index, for the case where there is no address specified at all.

Reply:
Actually no I was thinking about this too narrowly. The issue is really that the condition in the validation rule in question is not sufficient. I think it make sense to replace it with a custom validator that fully mimics the front end validation for address/foreign address in the UI. I can just go ahead an include that in this PR.

Reply:
Ok I backtracked on this, because I think the front end validation for Address is wrong too, and so I don't feel like there is a good source of truth at the moment. Created a separate task. In the mean time I at least fixed the specific issue you pointed out in your original comment here:  https://github.com/SaraAlert/SaraAlert/pull/743/commits/d7df34ce259dcad1e1c5d4dcc14735e8acd4f1b6 @ngfreiter

Reply:
Thanks for the fix. I'm fine with keeping it as is until we get a more solid source of truth.